### PR TITLE
Updates for v3.23.0

### PR DESCRIPTION
**Pull request by mgleung**
## Description

Updates for releasing v3.23.0

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+The Calico VPP dataplane integration is now in beta!
+
+#### Calico networking support in AKS
+
+You can now use the Calico CNI in your AKS clusters to take advantage of all of the Calico networking features.
+
+Pull Requests:
+
+ - Updated Calico doc for AKS Bring Your Own CNI. [calico #5857](https://github.com/projectcalico/calico/pull/5857) (@song-jiang)
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:23]: "Bring Your Own" should be "bring your own"

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ - Add IPv6 VXLAN route calculation support to Felix [calico #5919](https://github.com/projectcalico/calico/pull/5919) (@coutinhop)
+ - Allocate IPv6 VXLAN tunnel addresses [calico #5893](https://github.com/projectcalico/calico/pull/5893) (@coutinhop)
+
+#### VPP dataplane beta
+
+The Calico VPP dataplane integration is now in beta!
+
+#### Calico networking support in AKS
+
+You can now use the Calico CNI in your AKS clusters to take advantage of all of the Calico networking features.
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:19]: "Calico networking" instead of "the Calico CNI"

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+Pull Requests:
+
+ - Add BGP configuration options for turning off BGP loop prevention for eBGP peers. [calico #5736](https://github.com/projectcalico/calico/pull/5736) (@mgleung)
+ - Add passive configuration for global BGP peers [calico #5625](https://github.com/projectcalico/calico/pull/5625) (@mgleung)
+ - Add maxRestartTime and BGP password fields to BGP configurations for full mesh configurations. [calico #5609](https://github.com/projectcalico/calico/pull/5609) (@mgleung)
+
+#### Container Storage Interface (CSI) support
+
+Following the Kubernetes move from FlexVolumes, Calico now has its own CSI driver. Features that used to depend on FlexVolumes have been migrated to instead use CSI.
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:37]: Probably want to call out the upgrade / install implications for ALP more specifically and in more detail here.

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+Pull Requests:
+
+ - Replace the use of the flexvolume driver with a CSI driver [calico #5863](https://github.com/projectcalico/calico/pull/5863) (@mgleung)
+
+#### Windows HostProcess Containers Support (Tech Preview)
+
+You can now install Calico on Windows HostProcess containers. With HostProcess containers, you can manage your Calico installation using Kubernetes resources, such as DaemonSets and ConfigMaps.
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:45]: Would be nice to have a link to the docs for users to try this (and other features, where applicable)

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+ - ebpf: pod can connect to self when CTLB is disabled. [calico #5393](https://github.com/projectcalico/calico/pull/5393) (@tomastigera)
+ - FELIX_BPFEnforceRPF=Disabled/Strict (Strict by default) enforces strict RPF regardless of device settings in BPF mode. [calico #5342](https://github.com/projectcalico/calico/pull/5342) (@tomastigera)
+
+##### helm chart
+
+ - The tigera-operator helm chart now requires the user pre-create the `tigera-operator` namespace, or pass the `--create-namespace` option to `helm install`. Additionally, there are specific upgrade steps that MUST be performed when upgrading a helm install to Calico v3.23. Please see [the helm upgrade documentation](https://projectcalico.docs.tigera.io/maintenance/kubernetes-upgrade#upgrading-an-installation-that-was-installed-using-helm) for instructions. [calico #5649](https://github.com/projectcalico/calico/pull/5649) (@caseydavenport)
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:139]: Wonder if we should have a "Upgrade instructions" section or something that calls out this as well as the CSI work (I think those are the only two with any upgrade steps?) 

Probably worth updating our template to have a "Breaking changes" or "Require user action" section at the very top for important notices like this.

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+Pull Requests:
+
+ - Add IPv6 VXLAN support [calico #5988](https://github.com/projectcalico/calico/pull/5988) (@coutinhop)
+ - Add IPv6 VXLAN route calculation support to Felix [calico #5919](https://github.com/projectcalico/calico/pull/5919) (@coutinhop)
+ - Allocate IPv6 VXLAN tunnel addresses [calico #5893](https://github.com/projectcalico/calico/pull/5893) (@coutinhop)
+
+#### VPP dataplane beta
+
+The Calico VPP dataplane integration is now in beta!
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:15]: "The Calico VPP dataplane integration has reached beta status! Over the last few releases, the VPP team has made great strides in increasing stability, performance and feature compatibility."

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+You can now use the Calico CNI in your AKS clusters to take advantage of all of the Calico networking features.
+
+Pull Requests:
+
+ - Updated Calico doc for AKS Bring Your Own CNI. [calico #5857](https://github.com/projectcalico/calico/pull/5857) (@song-jiang)
+
+#### BGP enhancements
+
+We have added additional configuration options to allow for more control in specific scenarios, such as node-to-node mesh and separate systems running Calico using the same AS.
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:27]: > such as node-to-node mesh and separate systems running Calico using the same AS.

This makes it sounds like we introduced a mesh mode, as well as the ability to use different AS numbers.

We should spell out the specific changes and use-cases for them in more detail. There's no problem being a bit wordy here - we want to explain the value of what we did.

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+ - Helm chart allows configuration of tigera-operator resource requirements. [calico #5934](https://github.com/projectcalico/calico/pull/5934) (@sriramy)
+ - Add defensive checks to CNI plugin installer to improve reliability. [calico #5905](https://github.com/projectcalico/calico/pull/5905) (@caseydavenport)
+ - Increase Typha's "max fall behind" timeout to 300s.  This means that Typha will keep up to 300s of history if needed instead of 90s (thus increasing RAM usage if clients are falling a bit behind) but it reduces the chance of disconnecting a client spuriously in a very large cluster. [calico #5885](https://github.com/projectcalico/calico/pull/5885) (@fasaxc)
+ - Add Felix configuration option for programming floating IPs. [calico #5861](https://github.com/projectcalico/calico/pull/5861) (@caseydavenport)
+ - Produce both static and dynamic builds of the Calico CNI plugin, fixes compatibility with systems that do not include glibc. [calico #5826](https://github.com/projectcalico/calico/pull/5826) (@caseydavenport)
+ - Build components with go 1.17.8 to pick up CVE fixes. [calico #5817](https://github.com/projectcalico/calico/pull/5817) (@doublek)
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:101]: This was superceded by the 1.17.9 bullet point above:

>  - go version update to 1.17.9 [calico #6002](https://github.com/projectcalico/calico/pull/6002) (@doublek)


We can just remove this one.

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+ - Update the ipset package from 7.1 to 7.11 [calico #5485](https://github.com/projectcalico/calico/pull/5485) (@song-jiang)
+ - Add DataplaneWatchdogTimeout configuration parameter to Felix to allow the watchdog timeout to be increased.  Prevents spurious liveness failures on heavily loaded systems or systems operating at very high pod scale.  Increase default timeout to 90s (from 20s) after seeing too many issues where the liveness check failed spuriously. [calico #5484](https://github.com/projectcalico/calico/pull/5484) (@fasaxc)
+ - Add liveness callback to routing table updates to ensure watchdog doesn't time out when handling large numbers of updates. [calico #5483](https://github.com/projectcalico/calico/pull/5483) (@fasaxc)
+ - Downgrade watchercache.go log to debug level to reduce log spam [calico #5470](https://github.com/projectcalico/calico/pull/5470) (@caseydavenport)
+ - Rev up etcd to 3.5.1, any project imported calico should upgrade their etcd to over 3.5.
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:116]: This release note could use some better formatting. (it spans a few lines)

File: calico/_includes/release-notes/v3.23.0-release-notes.md (hunk 4 above)
Comment (lmm) [reply calico/_includes/release-notes/v3.23.0-release-notes.md:45]: https://projectcalico.docs.tigera.io/master/getting-started/windows-calico/quickstart#install-calico-for-windows-using-hostprocess-containers

Comment (lmm) [reply calico/_includes/release-notes/v3.23.0-release-notes.md:45]: How about:
> You can now install Calico for Windows using [HostProcess containers](https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod/).

Not entirely sure about the external link; we don't usually add them in release notes. Thought it might be nice in case the reader wants to learn more about HostProcess containers.

Comment (lmm) [reply calico/_includes/release-notes/v3.23.0-release-notes.md:45]: ^ Along the same lines, might be useful to add a link to the BYOCNI docs

File: calico/_includes/release-notes/v3.23.0-release-notes.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+ - Produce both static and dynamic builds of the Calico CNI plugin, fixes compatibility with systems that do not include glibc. [calico #5826](https://github.com/projectcalico/calico/pull/5826) (@caseydavenport)
+ - Add Felix configuration option to disable RPF check for specific workloads. [calico #5742](https://github.com/projectcalico/calico/pull/5742) (@AloysAugustin)
+ - Felix now supports setting the source address hint for workload routes separately for both IPv4 and IPv6 via the DeviceRouteSourceAddress and DeviceRouteSourceAddressIPv6 config params.  This allows for the IP address that kubelet uses to connect to a local pod to be controlled. [calico #5728](https://github.com/projectcalico/calico/pull/5728) (@hanamantagoudvk)
+ - New per-pool IPAM metrics added [calico #5706](https://github.com/projectcalico/calico/pull/5706) (@pasanw)
+ - Move PodDisruptionBudget from apiVersion policy/v1beta1 to policy/v1 [calico #5667](https://github.com/projectcalico/calico/pull/5667) (@coutinhop)
+ - Update flannel to version v0.16.3 [calico #5578](https://github.com/projectcalico/calico/pull/5578) (@cyclinder)
```
Comment (caseydavenport) [inline calico/_includes/release-notes/v3.23.0-release-notes.md:117]: This release note is actually wrong - looks like we're on v0.15.1

**Review by caseydavenport — APPROVED** [review]
